perception_and_utils_root/perception_and_utils/perception/detector_wrappers/object_detector.py

```diff
@@ -223,7 +226,9 @@ def process_frame(self, img_frame: np.ndarray) -> Tuple[np.ndarray, Dict[str, An
         )
         return updated_img_frame, object_scores
 
-    def process_frame_offline(self, img_frame: np.ndarray):
+    def process_frame_offline(
+        self, frame: DataFrame
+    ) -> Tuple[np.ndarray, Dict[str, Any]]:
```

Comment:
What was the difference b/w `process_frame` and `process_frame_offline`?

Reply:
```
            # object-detection + checking for intersection b/w all objects and hand
            # is a computationally expensive operation (happens inside
            # _get_scored_object_detections()).
            # For efficiency, in this code, we turn off object detection for the objects
            # that have already been detected and their place location has been recorded
            # This is done by removing the object_name from the labels and regenerating
            # the prompts for the object detector.
```

In the offline version, we disable the `_get_scored_object_detections` which is computationally expensive for detected and processed objects